### PR TITLE
Add jobs/{id}/builds route

### PR DESCRIPTION
**Pull request by d2lam**
- This uses the `getBuilds` method from the models
https://github.com/screwdriver-cd/models/pull/98

- Add test for pipeline create for coverage

- This will break until https://github.com/screwdriver-cd/datastore-dynamodb/pull/19 is merged.

Solves https://github.com/screwdriver-cd/screwdriver/issues/187

### CODE REVIEW

File: plugins/jobs/listBuilds.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+              .catch(err => reply(boom.wrap(err)));
+        },
+        response: {
+            schema: listSchema
+        },
+        validate: {
+            query: {
+                page: joi.number().integer().positive(),
+                count: joi.number().integer().positive(),
+                sort: joi.string().lowercase().valid(['ascending', 'descending'])
```
Comment (minz1027) [inline plugins/jobs/listBuilds.js]: Would it be better to re-use part of the scan schema? In case later on there are changes on the scan schema.

Comment (d2lam) [inline plugins/jobs/listBuilds.js]: Yeah, let me modify the data-schema first.

File: package.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
     "screwdriver-datastore-imdb": "^1.0.0",
     "screwdriver-executor-k8s": "^7.0.0",
     "screwdriver-executor-l3l": "^1.0.0",
-    "screwdriver-models": "^10.2.0",
+    "screwdriver-models": "^10.3.0",
```
Comment (FenrirUnbound) [inline package.json]: The `^` means it pulls in the latest minor version

Comment (stjohnjohnson) [inline package.json]: True, but the .3 requires at LEAST that version.

File: plugins/jobs/listBuilds.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+              })
+              .then(builds => reply(builds.map(b => b.toJson())))
+              .catch(err => reply(boom.wrap(err)));
+        },
+        response: {
+            schema: listSchema
+        },
+        validate: {
+            query: {
+                page: joi.number().integer().positive(),
```
Comment (stjohnjohnson) [inline plugins/jobs/listBuilds.js]: Isn't there already a data-schema for this?

File: plugins/jobs/listBuilds.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+            return factory.get(request.params.id)
+              .then(job => {
+                  if (!job) {
+                      throw boom.notFound('Job does not exist');
+                  }
+
+                  let config = {};
+
+                  if (Object.keys(request.query).length !== 0) {
```
Comment (stjohnjohnson) [inline plugins/jobs/listBuilds.js]: We should never be doing things like this, instead we should be using the default feature of Joi.

File: plugins/jobs/listBuilds.js (hunk 3 above)
Comment (d2lam) [inline plugins/jobs/listBuilds.js]: Yes, but it was exposed as an object with page and count. You can't access the page or count individually. I modified it so it has sort in there: https://github.com/screwdriver-cd/data-schema/blob/master/api/pagination.js so I'm just gonna change it to use that now

File: package.json (hunk 2 above)
Comment (d2lam) [inline package.json]: Yup. Because 10.2 does not have `getBuilds()`.

**Comment by d2lam** [comment]
Still waiting for 10.3 to be published:
https://app.wercker.com/Screwdriver/models/runs

**Pull request merged**